### PR TITLE
icloudpd: 1.24.0 ->  1.24.3

### PR DESCRIPTION
**Pull request by jnsgruk**
## Things done

Bumped `icloudpd` to [1.24.3](https://github.com/icloud-photos-downloader/icloud_photos_downloader/releases/tag/v1.24.3) and also updated related Python lib `srp` to [1.0.22](https://github.com/cocagne/pysrp/compare/1.0.21...1.0.22)

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jnsgruk** [comment]
Result of `nixpkgs-review pr 353782` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>icloudpd</li>
    <li>icloudpd.dist</li>
    <li>python311Packages.digi-xbee</li>
    <li>python311Packages.digi-xbee.dist</li>
    <li>python311Packages.srp</li>
    <li>python311Packages.srp.dist</li>
    <li>python312Packages.digi-xbee</li>
    <li>python312Packages.digi-xbee.dist</li>
    <li>python312Packages.srp</li>
    <li>python312Packages.srp.dist</li>
  </ul>
</details>

**Review by anpin — APPROVED** [review]
LGTM

File: pkgs/by-name/ic/icloudpd/package.nix [hunk 1]
```diff
@@ -74,6 +74,9 @@ python3Packages.buildPythonApplication rec {
     substituteInPlace pyproject.toml \
       --replace "setuptools==69.0.2" "setuptools" \
       --replace "wheel==0.42.0" "wheel"
+
+    substituteInPlace src/foundation/__init__.py \
+      --replace "0.0.1" "${version}"
```
Comment (drupol) [inline pkgs/by-name/ic/icloudpd/package.nix:79]:
```suggestion
      --replace-fail "0.0.1" "${version}"
```

Comment (jnsgruk) [reply pkgs/by-name/ic/icloudpd/package.nix]: Done! Thanks.

File: pkgs/by-name/ic/icloudpd/package.nix [hunk 2]
```diff
@@ -72,8 +72,11 @@ python3Packages.buildPythonApplication rec {
 
   preBuild = ''
     substituteInPlace pyproject.toml \
-      --replace "setuptools==69.0.2" "setuptools" \
-      --replace "wheel==0.42.0" "wheel"
+      --replace-fail "setuptools==69.0.2" "setuptools" \
```
Comment (drupol) [inline pkgs/by-name/ic/icloudpd/package.nix:75]: Why not using pythonRelaxDeps ?

Comment (jnsgruk) [reply pkgs/by-name/ic/icloudpd/package.nix:75]: afaik it doesn't work well with the build system components, or at least in the other python packages I maintain, I've never made it work with `setuptools`.

**Review by anpin — APPROVED** [review]
lgtm

File: pkgs/development/python-modules/srp/default.nix [hunk 3]
```diff
@@ -8,14 +8,14 @@
 
 buildPythonPackage rec {
   pname = "srp";
-  version = "1.0.21";
+  version = "1.0.22";
   format = "setuptools";
```
Comment (dotlambda) [inline pkgs/development/python-modules/srp/default.nix:12]:
```suggestion
  pyproject = true;
```

**Pull request merged**